internal/providers/pluginfw/resources/volume/data_volumes.go

```diff
@@ -69,7 +66,6 @@ func (d *VolumesDataSource) Read(ctx context.Context, req datasource.ReadRequest
 	if err != nil {
 		if apierr.IsMissing(err) {
 			resp.State.RemoveResource(ctx)
-			return
```

Comment:
Data source shouldn't return if the resource is missing. It should error. This is the same behaviour in SDKv2